sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -0,0 +1,184 @@
+from itertools import product
+
+import numpy as np
+
+from .. import confusion_matrix
+from ...utils import check_matplotlib_support
+from ...utils.multiclass import unique_labels
+from ...base import is_classifier
+
+
+class ConfusionMatrixDisplay:
+    """Confusion Matrix visualization.
+
+    It is recommend to use :func:`~sklearn.metrics.plot_confusion_matrix` to
+    create a :class:`ConfusionMatrixDisplay`. All parameters are stored as
+    attributes.
+
+    Read more in the :ref:`User Guide <visualizations>`.
+
+    Parameters
+    ----------
+    confusion_matrix : ndarray of shape (n_classes, n_classes)
+        Confusion matrix.
+
+    normalize : bool
+        Confusion matrix is normalized.
+
+    target_names : ndarray of shape (n_classes,)
+        Target names.
+
+    Attributes
+    ----------
+    im_: matplotlib AxesImage
+        Image representing the confusion matrix.
+
+    text_: ndarray of shape (n_classes, n_classes), dtype=matplotlib Text, or \
+           `None`
+        Array of matplotlib axes. `None` if `include_values` is false.
+
+    ax_ : matplotlib Axes
+        Axes with confusion matrix.
+
+    figure_ : matplotlib Figure
+        Figure containing the confusion matrix.
+    """
+    def __init__(self, confusion_matrix, normalize, target_names):
+        self.confusion_matrix = confusion_matrix
+        self.normalize = normalize
+        self.target_names = target_names
+
+    def plot(self, include_values=True, cmap='viridis', xticks_rotation=0.0,
+             ax=None):
+        """Plot visualization.
+
+        Parameters
+        ----------
+        include_values : bool, default=True
+            Includes values in confusion matrix.
+
+        cmap : str or matplotlib Colormap, default='viridis'
+            Colormap recognized by matplotlib.
+
+        xticks_rotation: float, default=0.0
+            Rotation of xtick labels.
+
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.ConfusionMatrixDisplay`
+        """
+        check_matplotlib_support("ConfusionMatrixDisplay.plot")
+        import matplotlib.pyplot as plt
+
+        if ax is None:
+            fig, ax = plt.subplots()
+        else:
+            fig = ax.figure
+
+        cm = self.confusion_matrix
+        self.im_ = ax.imshow(cm, interpolation='nearest', cmap=cmap)
+        self.text_ = None
+
+        if include_values:
+            self.text_ = np.empty_like(cm, dtype=object)
+            fmt = '.2f' if self.normalize else 'd'
+            thresh = cm.max() / 2.
+            for i, j in product(range(cm.shape[0]), range(cm.shape[1])):
+                color = "white" if cm[i, j] < thresh else "black"
```

Comment:
I think that's weird as it doesn't depend on the colormap.
Here's how I usually do it:
https://github.com/amueller/mglearn/blob/master/mglearn/tools.py#L76

without depending on the colormap there's no way this works, right? because someone could use ``greys`` and ``greys_r`` and they clearly need the opposite colors.

Reply:
I think it should be pcolormesh not pcolor, though.

Reply:
also: shouldn't this go in a separate helper function? It's probably not the only time we want to show a heatmap (grid search will need this as well)

Reply:
For reference: https://matplotlib.org/3.1.1/gallery/images_contours_and_fields/image_annotated_heatmap.html#sphx-glr-gallery-images-contours-and-fields-image-annotated-heatmap-py

Updated PR, with an alternative: it uses the colormap to get the colors for the text.

Reply:
Looks reasonable.
Can you maybe add a test? Like calling ``ConfusionMatrixDisplay`` with ``np.eye(2)`` and ``plt.cm.greys`` and check that the text colors are black white black white and with ``plt.cm.greys_r`` and check that the text colors are white black white black?